src/libra_py/CP2K_methods.py

```diff
@@ -197,55 +199,354 @@ def read_cp2k_tddfpt_log_file( tddfpt_log_file_name, num_ci_states, tolerance ):
                 # When found the line in which contains 'Excitation analysis' in 
                 # the log file, append it in the variable exc_anal_line
                 exc_anal_line = i
-        
+
         if 'states' in lines[i].lower().split():
             if 'multiplicity' in lines[i].lower().split():
 
-                # Here we search for the line that contains
-                # 'R-TDDFPT states of multiplicity 1' in the log file
+                # Here we search for the line that contains 'R-TDDFPT states of multiplicity 1'
+                # or 'U-TDDFPT states of multiplicity 1' in the log file
                 r_tddfpt_line = i
 
     excitation_energies = []
-    
+
     # Start from 5 lines after finding the line contaning 'R-TDDFPT states of multiplicity 1'
     # This is because they contain the energies from that line.
     for i in range( r_tddfpt_line+5, len( lines ) ):
         if len( lines[i].split() ) == 0:
             break
         excitation_energies.append( float( lines[i].split()[2] ) )
-        
+
     # Start from 5 lines after finding the line contaning 'Excitation analysis'
     # From that point we have the state numbers with their configurations.
     # So, we append the lines which contain only 'State number' and stop
     # whenever we reach to a blank line.
     state_num_lines = []
     for i in range( exc_anal_line+5, len( lines ) ):
 
-        if len( lines[i].split() ) == 0:
-            state_num_lines.append( i )
-            break
+        if isUKS == 1:
+
+            if len( lines[i].split() ) == 0 or '----' in lines[i]:
```

Comment:
in this section, do
 tmp = lines[i].split()  then use tmp as needed; otherwise, every call of lines[i].slipt() takes some extra time since it is a function call, so it is not efficient